### PR TITLE
Flush uyuni roster cache if the config has changed

### PR DESCRIPTION
**Pull request by vzhestkov**
## What does this PR change?

Improves the config management for `uyuni` roster module for salt with updating the cached data on changing the config for uyuni roster module.

In some cases if the pre flight script was disabled and then enabled back the cached roster data will not use the pre flight script.
The opposite situation is also possible.

## GUI diff

No difference.

## Documentation
- No documentation needed: only internal and user invisible changes

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Review by SchoolGuy — APPROVED** [review]
Change looks good to me!

File: susemanager-utils/susemanager-sls/modules/roster/uyuni.py [hunk 1]
```diff
@@ -62,6 +62,9 @@ class UyuniRoster:
     """
 
     def __init__(self, db_config, uyuni_roster_config):
+        self.config_hash = hashlib.sha256(
+            str(uyuni_roster_config).encode(errors="backslashreplace")
```
Comment (agraul) [inline susemanager-utils/susemanager-sls/modules/roster/uyuni.py:66]: What's the type of `uyuni_roster_config`?

Comment (vzhestkov) [reply susemanager-utils/susemanager-sls/modules/roster/uyuni.py:66]: It's a dictionary.

Comment (agraul) [reply susemanager-utils/susemanager-sls/modules/roster/uyuni.py:66]: Ah, ok. The order of the keys is stable, right?

Comment (SchoolGuy) [reply susemanager-utils/susemanager-sls/modules/roster/uyuni.py:66]: Order of keys is only stable from Python 3.6 onwards: https://docs.python.org/3/library/stdtypes.html#dict

Comment (SchoolGuy) [reply susemanager-utils/susemanager-sls/modules/roster/uyuni.py:66]: But I guess we only have Python 3.6+ anyway, so we should be safe.

Comment (vzhestkov) [reply susemanager-utils/susemanager-sls/modules/roster/uyuni.py:66]: Right, 3.6 is the only version we are using for Uyuni/4.2/4.1 on the master side where this module could be invoked.

Comment (agraul) [reply susemanager-utils/susemanager-sls/modules/roster/uyuni.py:66]: I was actually thinking about whatever code path adds the stuff to the dict (let's image something parses a YAML file and randomly sorts it before adding the contents to `uyuni_roster_config`), but I agree - the worst case is not that bad anyway.

Comment (vzhestkov) [reply susemanager-utils/susemanager-sls/modules/roster/uyuni.py:66]: The source of this dict actually a `__opts__` from the master (the same for salt-api): `__opts__.get("uyuni_roster")`
It's defined in the master's config.

**Pull request merged**